src/angleaxis_types.jl

```diff
@@ -84,7 +84,7 @@ end
     theta = 2 * atan(sin_t2, q.w)
     num_pert = eps(typeof(theta))^2
```

Comment:
Thanks for catching the unnecessary normalization in `principal_value`. If you want to remove the normalization in the `Quat` --> `AngleAxis` conversion I think that you should change the exponent of `eps` to `4`. If you don't, I think that that axis will not be a unit vector for things like `Quat(1.0, 0.0, 1.0e-20, 0.0)`. I think that you have a test in `RigidBodyDynamics` that cares about the behavior of this function for `Quat` real parts below `eps`.

Reply:
Wow, thanks for the heads up!

Reply:
With the exponent `=2`, the axis will start coming unnormalized at around eps (see below):

`Quat(1.0, 0.0, 2.2204460492503155e-16, 0.0) --> norm(axis) = 0.9999999999999996`
`Quat(1.0, 0.0, 2.7105054312137596e-20, 0.0) --> norm(axis) = 0.9999999999963618`
`Quat(1.0, 0.0, 3.3087224502121305e-24, 0.0) --> norm(axis) = 0.9999999701976785`
`Quat(1.0, 0.0, 4.0389678347315975e-28, 0.0) --> norm(axis) = 0.9997559189687326`
**`Quat(1.0, 0.0, 4.930380657631336e-32, 0.0) --> norm(axis) = 0.4999999999999999`**
`Quat(1.0, 0.0, 6.018531076210116e-36, 0.0) --> norm(axis) = 0.9997559189687326`
`Quat(1.0, 0.0, 7.34683969263941e-40, 0.0) --> norm(axis) = 0.9999999701976785`
`Quat(1.0, 0.0, 8.96831017167895e-44, 0.0) --> norm(axis) = 0.999999999996362`
`Quat(1.0, 0.0, 1.0947644252537762e-47, 0.0) --> norm(axis) = 0.9999999999999996`

If the exponent is pushed to `4` then axis will start coming unnormalized at around eps^2 which is a number I am more comfortable with.

I bring this up in because rotation conversion behavior (generally) comes up in places. E.g. https://github.com/JuliaRobotics/RigidBodyDynamics.jl/blob/c425c6cfbc860588388f50bc7ef8498bc836921b/test/test_spatial.jl#L234

Reply:
The only disadvantage to `4` instead of `2` is making the commit... and you already paid that price.